catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -258,8 +274,8 @@ private boolean orderContainsSupportedDelivery() {
         if (order.del_list != null) {
             for (DeliveryDetails deliveryDetails : order.del_list) {
                 Destination destination = deliveryDetails.dests;
-                if (destination.discriminator() == DestinationType.FTP
-                        && destination.f_dest() != null) {
+                if (((destination.discriminator() == DestinationType.FTP) && (destination.f_dest() != null)) ||
+                        ((destination.discriminator() == DestinationType.EMAIL) && (destination.e_dest() != null))) {
```

Comment:
You might consider a streaming solution. For complex cases like this, the declarative nature can be much more easily read:
```java
return order.del_list != null && Arrays.stream(order.del_list)
        .map(d -> d.dests)
        .filter(d -> d.discriminator() == DestinationType.FTP)
        .filter(d -> d.f_dest() != null)
        .filter(d -> d.discriminator() == DestinationType.EMAIL)
        .filter(d -> d.e_dest() != null)
        .anyMatch(Objects::nonNull);
```